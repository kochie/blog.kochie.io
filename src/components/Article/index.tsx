import React, { PropsWithChildren } from 'react'
import Link from 'next/link'
import Jumbotron from '@/components/Jumbotron'
import Card from '@/components/Card'
import { Tag, TagSet } from '@/components/Tag'
import Image from 'next/image'

// eslint-disable-next-line import/no-unresolved
// import { Article as ArticleDetails } from 'articles.json'
// eslint-disable-next-line import/no-unresolved
// import { Author as AuthorDetails } from 'authors.json'

import style from './Article.module.css'
import { ArticleMetadata } from 'src/lib/article-path'
import { Author } from 'metadata.yaml'

interface ArticleProps {
  article: ArticleMetadata
  author: Author
}

interface AuthorLinkProps {
  username: string
  fullname: string
}

const Article = ({
  article,
  author,
  children,
}: PropsWithChildren<ArticleProps>): React.ReactElement => {
  const AuthorLink = ({
    username,
    fullname,
  }: AuthorLinkProps): React.ReactElement => {
    const link = (
      <Link href={'/authors/[authorId]'} as={`/authors/${username}`}>
        <a className={style.underline}>{fullname}</a>
      </Link>
    )
    return (
      <>
        <p>Written by {link}</p>
      </>
    )
  }

  // console.log(decodeBlurHash(article.jumbotron.lqip))

  return (
    <>
      <Jumbotron
        height={'60vh'}
        width={'100vw'}
        background={
          <div className="">
            <Image
              src={article.jumbotron.url}
              layout="fill"
              blurDataURL={article.jumbotron.lqip}
              objectFit="cover"
              objectPosition="center"
              placeholder="blur"
            />
          </div>
        }
        foreground={<div className="h-full w-full overflow-hidden" />}
      />
      <div className="relative max-w-5xl -mt-20 mx-auto px-4 mb-0 pb-10">
        <div>
          <Card>
            <div className="p-4 md:p-8 lg:p-14">
              <h1 className="my-3 text-3xl md:text-left text-center">
                {article.title}
              </h1>
              <TagSet className="md:mb-3 md:-ml-1 justify-center md:justify-start">
                {article.tags.map((tag) => (
                  <Tag name={tag} link={`/tags/${tag}`} key={tag} inverted />
                ))}
              </TagSet>
              <div className="flex flex-col md:flex-row justify-between items-center">
                <div>
                  <span className={style.subText}>
                    {`Published on ${new Date(
                      article.publishedDate
                    ).toLocaleDateString('en')}`}
                  </span>
                </div>
                <div>
                  {article.editedDate == article.publishedDate ? null : (
                    <span className={style.subText}>
                      {`Last edited on ${new Date(
                        article.editedDate
                      ).toLocaleDateString('en')}`}
                    </span>
                  )}
                </div>
              </div>
              <div className="flex flex-col md:flex-row justify-between items-center">
                <span className={style.subText}>
                  <AuthorLink
                    username={author.username}
                    fullname={author.fullName}
                  />
                </span>
                <div>
                  <span className={style.subText}>{article.readTime}</span>
                </div>
              </div>
              <div style={{ marginTop: '5px' }}></div>
              {children}
            </div>
          </Card>
        </div>
      </div>
    </>
  )
}

export default Article
