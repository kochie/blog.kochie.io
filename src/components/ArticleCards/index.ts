export interface CardDetails {
  title: string
  image: {
    lqip: string
    alt: string
    url: string
  }
  blurb: string
  readTime: string
  tags: string[]
  articleDir: string
}

export { default as SmallCard } from './Small'
export { default as MediumCard } from './Medium'
export { default as LargeCard } from './Large'

import SmallCard from './Small'
import MediumCard from './Medium'
import LargeCard from './Large'

const Cards = { Small: SmallCard, Medium: MediumCard, Large: LargeCard }

export default Cards
