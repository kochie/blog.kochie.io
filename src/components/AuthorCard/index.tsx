import Image from 'next/image'
import Link from 'next/link'
import { Author } from 'types/metadata'
import Card from '../Card'
import SMButton from '../SocialMediaButtons'

import styles from '../../styles/list.module.css'

export function AuthorCardLeft({ author }: { author: Author }) {
  return (
    <div className="relative max-w-5xl mx-auto px-4 mb-0 pb-10 mt-10">
      <Card>
        <div className="px-4 md:px-8 lg:px-14 py-5 flex items-center flex-col justify-center md:justify-start md:flex-row group">
          <Link href={`/authors/${author.username}`}>
            <div className="w-32 h-32 relative border-4 border-white border-solid rounded-full md:mr-4 overflow-hidden">
              <div className="relative transition ease-in-out duration-500 filter grayscale-70 group-hover:grayscale-0 w-full h-full">
                <Image
                  width={128}
                  height={128}
                  sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
                  src={`/images/authors/${author.avatar.src}`}
                  alt={`${author.fullName} Avatar`}
                  placeholder="blur"
                  blurDataURL={author.avatar.lqip || ''}
                  className="transform-gpu group-hover:scale-110 flex-shrink-0 cursor-pointer transition ease-in-out duration-500"
                />
              </div>
            </div>
          </Link>

          <div className="m-4">
            <div className="flex-wrap flex flex-col md:flex-row items-center text-2xl">
              <h1 className={styles.heading}>
                <Link href={`/authors/${author.username}`}>
                  {author.fullName}
                </Link>
              </h1>
              <div className="flex md:ml-4 md:my-0 my-2 gap-1">
                {author.socialMedia.map((sm) => (
                  <SMButton sm={sm} key={sm.name} />
                ))}
              </div>
            </div>
            <p className="text-center md:text-left mt-2">{author.bio}</p>
          </div>
        </div>
      </Card>
    </div>
  )
}

export function AuthorCardRight({ author }: { author: Author }) {
  return (
    <Card key={author.username}>
      <div className="p-5 flex items-center flex-col justify-start md:flex-row-reverse group">
        <Link href={`/authors/${author.username}`}>
          <div className="w-32 h-32 relative border-4 border-white border-solid rounded-full ml-4 overflow-hidden">
            <div className="transition ease-in-out duration-500 filter grayscale-70 group-hover:grayscale-0 w-full h-full">
              <Image
                width={128}
                height={128}
                sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
                src={`/images/authors/${author.avatar.src}`}
                alt={`${author.fullName} Avatar`}
                placeholder="blur"
                blurDataURL={author.avatar.lqip || ''}
                className="transform-gpu group-hover:scale-110 flex-shrink-0 transition ease-in-out duration-500 filter grayscale-70 cursor-pointer group-hover:grayscale-0"
              />
            </div>
          </div>
        </Link>

        <div className="m-4">
          <div className="flex-wrap flex flex-col md:flex-row-reverse items-center text-2xl">
            <h1 className={styles.heading}>
              <Link href={`/authors/${author.username}`}>
                {author.fullName}
              </Link>
            </h1>
            <div className="flex md:mr-4 md:my-0 my-2">
              {author.socialMedia.map((sm) => (
                <SMButton sm={sm} key={sm.name} />
              ))}
            </div>
          </div>
          <p className="text-center md:text-right">{author.bio}</p>
        </div>
      </div>
    </Card>
  )
}
