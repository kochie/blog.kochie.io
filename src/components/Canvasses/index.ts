export { default as HaloInteractive } from './ring-spinner'
