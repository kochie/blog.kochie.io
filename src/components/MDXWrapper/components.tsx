import Image from 'next/image'
import Link from 'next/link'
import {
  IframeHTMLAttributes,
  PropsWithChildren,
  ReactElement,
  ReactNode,
} from 'react'
import { TOC } from '@/lib/rehype-toc-plugin'
import CodeBlock from '../CodeBlocks'
import GithubProject from '../GithubProject'
import Quote from '../Quote'
import HaloInteractive from '../Canvasses/ring-spinner'
// import type { StandardLonghandProperties } from 'csstype'

type PropsOnlyChildren = {
  children?: ReactNode
}

interface HeadingProps {
  id?: string
}

const PRE = ({ children }: PropsOnlyChildren) => {
  if (!children) return <pre />
  if (
    typeof children === 'string' ||
    typeof children === 'number' ||
    typeof children === 'boolean'
  ) {
    return <pre>{children}</pre>
  }

  // tried this a few times, but it doesn't work
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-expect-error
  const className = children.props.className
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-expect-error
  const grandChildren = children.props.children
  return <CodeBlock className={className}>{grandChildren}</CodeBlock>
}

const H1 = ({
  children,
  id,
}: PropsWithChildren<HeadingProps>): ReactElement => (
  <h1 className="text-5xl my-8" style={{ scrollMarginTop: '50px' }} id={id}>
    {children}
  </h1>
)

const H2 = ({
  children,
  id,
}: PropsWithChildren<HeadingProps>): ReactElement => (
  <h2 className="text-3xl my-8" style={{ scrollMarginTop: '50px' }} id={id}>
    {children}
  </h2>
)

const H3 = ({
  children,
  id,
}: PropsWithChildren<HeadingProps>): ReactElement => {
  return (
    <h3 className="text-2xl my-8" style={{ scrollMarginTop: '50px' }} id={id}>
      {children}
    </h3>
  )
}

const H4 = ({
  children,
  id,
}: PropsWithChildren<HeadingProps>): ReactElement => {
  return (
    <h4 className="text-xl my-8" style={{ scrollMarginTop: '50px' }} id={id}>
      {children}
    </h4>
  )
}

const H5 = ({
  children,
  id,
}: PropsWithChildren<HeadingProps>): ReactElement => {
  return (
    <h5 className="text-lg my-8" style={{ scrollMarginTop: '50px' }} id={id}>
      {children}
    </h5>
  )
}

const H6 = ({
  children,
  id,
}: PropsWithChildren<HeadingProps>): ReactElement => {
  return (
    <h6 className="text-base my-8" style={{ scrollMarginTop: '50px' }} id={id}>
      {children}
    </h6>
  )
}

const IMG = ({
  src,
  alt,
  lqip,
}: {
  src?: string
  alt?: string
  lqip?: string
  articleDir?: string
}): ReactElement => {
  const params = new URLSearchParams(src?.split('?')[1])
  const filename = src?.split('?')[0] ?? ''

  let image

  if (filename.endsWith('.svg')) {
    image = (
      <div className="relative rounded-t-xl overflow-hidden">
        <picture>
          <source srcSet={src} type="image/svg+xml" />
          <img
            src={src}
            alt={alt}
            width={parseInt(params.get('width') ?? '0')}
            height={parseInt(params.get('height') ?? '0')}
          />
        </picture>
      </div>
    )
  } else {
    image = (
      <div className="relative w-full h-auto rounded-t-xl overflow-hidden">
        <Image
          src={src || ''}
          sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
          style={{
            // width: '100%',
            height: 'auto',
            // @ts-expect-error - objectFit is not a valid property
            objectFit: params.get('objectFit') ?? undefined,
          }}
          placeholder="blur"
          blurDataURL={lqip}
          height={parseInt(params.get('height') ?? '0')}
          width={parseInt(params.get('width') ?? '0')}
          alt={alt ?? ''}
        />
      </div>
    )
  }

  return (
    <div className={`mx-auto my-10 w-fit`}>
      {image}
      <div className="rounded-b-xl bg-gray-700 text-sm text-white">
        <div className="p-4">{alt}</div>
      </div>
    </div>
  )
}

const Iframe = (
  props: IframeHTMLAttributes<HTMLIFrameElement>
): ReactElement => (
  <div className="w-full my-10">
    {/* eslint-disable-next-line @typescript-eslint/no-explicit-any */}
    <iframe className="mx-auto" {...(props as any)} />
  </div>
)

const P = ({ children }: PropsOnlyChildren): ReactElement => {
  if (typeof children === 'string') {
    return <p className="my-3">{children}</p>
  }
  return <div className="my-3">{children}</div>
}

const BLOCKQUOTE = ({ children }: PropsOnlyChildren) => (
  <blockquote className="bg-white px-8 py-2 my-5 rounded-lg text-black">
    {children}
  </blockquote>
)

const ANCHOR = ({
  children,
  href,
  ...props
}: PropsWithChildren<{ href?: string }>) => (
  <Link
    href={href ?? ''}
    {...props}
    className="underline font-bold scroll-my-14 dark:text-yellow-400 dark:hover:text-yellow-600 text-purple-600 hover:text-purple-800 duration-200"
  >
    {children}
  </Link>
)

const CODE = ({ children }: PropsWithChildren<Record<never, never>>) => (
  <code className="dark:bg-gray-800 text-sm p-1 rounded font-mono bg-gray-400 dark:text-white">
    {children}
  </code>
)

const OL = ({ children, id }: PropsWithChildren<{ id?: string }>) => (
  <ol id={id} className="list-decimal list-outside px-12">
    {children}
  </ol>
)

const UL = ({ children, id }: PropsWithChildren<{ id?: string }>) => (
  <ul id={id} className="list-inside">
    {children}
  </ul>
)

const LI = ({ children, id }: PropsWithChildren<{ id?: string }>) => (
  <li id={id} className="list-item">
    {children}
  </li>
)

const SUP = ({ children, id }: PropsWithChildren<{ id?: string }>) => (
  <sup id={id} style={{ scrollMarginTop: '50px' }}>
    {children}
  </sup>
)

const HR = () => <hr className="my-6 border-2 mx-8" />

export const components = {
  H1,
  H2,
  H3,
  H4,
  H5,
  H6,
  IMG,
  Iframe,
  P,
  BLOCKQUOTE,
  ANCHOR,
  CODE,
  OL,
  UL,
  LI,
  SUP,
  HR,
  PRE,
  GithubProject,
  Quote,
  HaloInteractive,
  TOC,
}
