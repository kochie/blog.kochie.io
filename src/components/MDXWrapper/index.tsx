'use client'

import { MDXRemote, type MDXRemoteProps } from 'next-mdx-remote'
import {
  H1,
  H2,
  H3,
  H4,
  H5,
  H6,
  IMG,
  LI,
  OL,
  P,
  UL,
  HaloInteractive,
  Iframe,
  GithubProject,
  BLOCKQUOTE,
  ANCHOR,
  CODE,
  TOC,
  SUP,
  HR,
  Quote,
} from './components'
import React from 'react'

// https://github.com/hashicorp/next-mdx-remote/blob/main/src/index.tsx
export const MDXContent: React.FC<MDXRemoteProps> = (props) => {
  return (
    <MDXRemote
      {...props}
      components={{
        h1: H1,
        h2: H2,
        h3: H3,
        h4: H4,
        h5: H5,
        h6: H6,
        img: IMG,
        p: P,
        ol: OL,
        ul: UL,
        li: LI,
        HaloInteractive,
        iframe: Iframe,
        GithubProject,
        blockquote: BLOCKQUOTE,
        a: ANCHOR,
        code: CODE,
        TOC,
        sup: SUP,
        hr: HR,
        Quote,
      }}
    />
  )
}
