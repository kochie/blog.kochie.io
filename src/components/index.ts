export { default as Topbar } from './Topbar'
export { default as Jumbotron } from './Jumbotron'
export { default as Gallery } from './Gallery'
export { default as Footer } from './Footer'
export { default as Article } from './Article'
export { Tag, TagSet } from './Tag'
export { default as Card } from './Card'
export { default as Page } from './Page'
export { default as Heading } from './Heading'
export { default as Loading } from './Loading'
export { HaloInteractive } from './Canvasses'
export { ThemeProvider, THEME, useTheme, ThemeButton } from './Theme'
export { default as Quote } from './Quote'
export { default as Revue } from './Revue'
export { default as CodeBlock } from './CodeBlocks'
export { default as GithubProject } from './GithubProject'
export { default as SMButton } from './SocialMediaButtons'
export { AuthorCardLeft, AuthorCardRight } from './AuthorCard'
export { default as TopButton } from './TopButton'
export { default as Title } from './Heading/title'
export { AppleLinks } from './Heading/apple'
export { Common } from './Heading/common'

export { default as ArticleCards } from './ArticleCards'
export * from './ArticleCards'
