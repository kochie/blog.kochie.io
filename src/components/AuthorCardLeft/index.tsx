import Image from 'next/image'
import Link from 'next/link'
import { Author } from 'types/metadata'
import { Card, SocialMediaButton } from '@/components'

import styles from '@/styles/list.module.css'

export default function AuthorCardLeft({ author }: { author: Author }) {
  return (
    <div className="relative max-w-5xl mx-auto px-4 mb-0 pb-10 mt-10">
      <Card>
        <div className="px-4 md:px-8 lg:px-14 py-5 flex items-center flex-col justify-center md:justify-start md:flex-row group">
          <Link href={`/authors/${author.username}`}>
            <div className="w-32 h-32 relative border-4 border-white border-solid rounded-full md:mr-4 overflow-hidden">
              <div className="relative transition ease-in-out duration-500 filter grayscale-70 group-hover:grayscale-0 w-full h-full">
                <Image
                  width={128}
                  height={128}
                  sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
                  src={`/images/authors/${author.avatar.src}`}
                  alt={`${author.fullName} Avatar`}
                  placeholder="blur"
                  blurDataURL={author.avatar.lqip || ''}
                  className="transform-gpu group-hover:scale-110 flex-shrink-0 cursor-pointer transition ease-in-out duration-500"
                />
              </div>
            </div>
          </Link>

          <div className="m-4">
            <div className="flex-wrap flex flex-col md:flex-row items-center text-2xl">
              <h1 className={styles.heading}>
                <Link href={`/authors/${author.username}`}>
                  {author.fullName}
                </Link>
              </h1>
              <div className="flex md:ml-4 md:my-0 my-2 gap-1">
                {author.socialMedia.map((sm) => (
                  <SocialMediaButton sm={sm} key={sm.name} />
                ))}
              </div>
            </div>
            <p className="text-center md:text-left mt-2">{author.bio}</p>
          </div>
        </div>
      </Card>
    </div>
  )
}
