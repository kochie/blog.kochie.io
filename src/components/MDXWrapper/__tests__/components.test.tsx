import { describe, test, expect } from 'vitest'
import { render } from '@testing-library/react'
import { components } from '../components'

describe('MDX COMPONENTS', () => {
  test('renders headings correctly', () => {
    const { asFragment } = render(
      <>
        <components.h1>Heading 1</components.h1>
        <components.h2>Heading 2</components.h2>
        <components.h3>Heading 3</components.h3>
        <components.h4>Heading 4</components.h4>
        <components.h5>Heading 5</components.h5>
        <components.h6>Heading 6</components.h6>
      </>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders paragraphs correctly', () => {
    const { asFragment } = render(<components.p>Paragraph</components.p>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders blockquote correctly', () => {
    const { asFragment } = render(
      <components.blockquote>Blockquote</components.blockquote>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders ul correctly', () => {
    const { asFragment } = render(<components.ul>Unordered List</components.ul>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders ol correctly', () => {
    const { asFragment } = render(<components.ol>Ordered List</components.ol>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders li correctly', () => {
    const { asFragment } = render(<components.li>List Item</components.li>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders hr correctly', () => {
    const { asFragment } = render(<components.hr />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders a correctly', () => {
    const { asFragment } = render(
      <components.a href="https://example.com">Link</components.a>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders code correctly', () => {
    const { asFragment } = render(<components.code>Strong</components.code>)
    expect(asFragment()).toMatchSnapshot()
  })
})
