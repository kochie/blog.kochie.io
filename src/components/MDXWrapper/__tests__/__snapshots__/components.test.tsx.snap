// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MDX COMPONENTS > renders a correctly 1`] = `
<DocumentFragment>
  <a
    class="underline font-bold scroll-my-14 dark:text-yellow-400 dark:hover:text-yellow-600 text-purple-600 hover:text-purple-800 duration-200"
    href="https://example.com"
  >
    Link
  </a>
</DocumentFragment>
`;

exports[`MDX COMPONENTS > renders blockquote correctly 1`] = `
<DocumentFragment>
  <blockquote
    class="bg-white px-8 py-2 my-5 rounded-lg text-black"
  >
    Blockquote
  </blockquote>
</DocumentFragment>
`;

exports[`MDX COMPONENTS > renders code correctly 1`] = `
<DocumentFragment>
  <code
    class="dark:bg-gray-800 text-sm p-1 rounded font-mono bg-gray-400 dark:text-white"
  >
    Strong
  </code>
</DocumentFragment>
`;

exports[`MDX COMPONENTS > renders headings correctly 1`] = `
<DocumentFragment>
  <h1
    class="text-5xl my-8"
    style="scroll-margin-top: 50px;"
  >
    Heading 1
  </h1>
  <h2
    class="text-3xl my-8"
    style="scroll-margin-top: 50px;"
  >
    Heading 2
  </h2>
  <h3
    class="text-2xl my-8"
    style="scroll-margin-top: 50px;"
  >
    Heading 3
  </h3>
  <h4
    class="text-xl my-8"
    style="scroll-margin-top: 50px;"
  >
    Heading 4
  </h4>
  <h5
    class="text-lg my-8"
    style="scroll-margin-top: 50px;"
  >
    Heading 5
  </h5>
  <h6
    class="text-base my-8"
    style="scroll-margin-top: 50px;"
  >
    Heading 6
  </h6>
</DocumentFragment>
`;

exports[`MDX COMPONENTS > renders hr correctly 1`] = `
<DocumentFragment>
  <hr
    class="my-6 border-2 mx-8"
  />
</DocumentFragment>
`;

exports[`MDX COMPONENTS > renders li correctly 1`] = `
<DocumentFragment>
  <li
    class="list-item"
  >
    List Item
  </li>
</DocumentFragment>
`;

exports[`MDX COMPONENTS > renders ol correctly 1`] = `
<DocumentFragment>
  <ol
    class="list-decimal list-outside px-12"
  >
    Ordered List
  </ol>
</DocumentFragment>
`;

exports[`MDX COMPONENTS > renders paragraphs correctly 1`] = `
<DocumentFragment>
  <p
    class="my-3"
  >
    Paragraph
  </p>
</DocumentFragment>
`;

exports[`MDX COMPONENTS > renders ul correctly 1`] = `
<DocumentFragment>
  <ul
    class="list-inside"
  >
    Unordered List
  </ul>
</DocumentFragment>
`;
